### ISSUE TITLE
Scaling, even integer scaling completely breaks rendering in firefox

### ISSUE THREAD

**Issue by Kommynct** [hyprwm/Hyprland#5153]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit 7587cadd0a4e94c9ba42c00307854052798cf9b2 dirty (renderer: add support for gles3.2 screen shaders).
Date: Mon Mar 18 00:15:04 2024
Tag: v0.37.1-11-g7587cadd

flags: (if any)


System Information:
System name: Linux
Node name: anarchy-desktop
Release: 6.8.1-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Sat, 16 Mar 2024 17:15:23 +0000


GPU information: 
06:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 23 [Radeon RX 6650 XT / 6700S / 6800S] [1002:73ef] (rev c1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:


```

</details>

### Bug or Regression?

Bug

### Description

Only the top left corner of my display is being rendered properly if i set scaling to 2

### How to reproduce

Take a monitor
set scaling to 2
notice only the top left of the screen is rendered at all

my full monitor settings in case that's needed:

monitor=HDMI-A-1,2560x1440@60,0x0,2
monitor=DP-1,2560x1440@143.856,1280x0,1
monitor=DP-2,1920x1080@60,3840x0,1

I also use kanshi to switch the monitor that is scaled off and on.

**Comment by vaxerski** [comment]
I use scaling normally on my laptop and it's working just fine?

Maybe try without kanshi?

I've seen a report of this once, but outside of that there are tons of people using scaling and I've never heard this happen outside of that.

**Comment by Kommynct** [comment]
I uninstalled kanshi but the issue persists, 

https://github.com/hyprwm/Hyprland/assets/13646790/5680250e-e2c7-425f-be32-73d691ee1d24

here's a video.

**Comment by Kommynct** [comment]
Ah, i've found the problem, long ago, in about:config I toggled 

widget.wayland.fractional-scale.enabled	true

If you enable that, and restart firefox, the issue is reproducible

**Comment by vaxerski** [comment]
IIRC firefox's fractional-scale is very buggy. That might be a ff issue.

**Comment by Kommynct** [comment]
I would agree, but I used the option in sway and never had any issues at all.

**Comment by vaxerski** [comment]
and chromium uses the same protocol and doesn't have issues either :)

**Comment by eddiezane** [comment]
>  2024-03-18.02-19-23.mp4 
> here's a video.

Seeing the same thing in Chrome after updating. I removed `--disable-features=WaylandFractionalScaleV1` from `~/.config/chrome-flags.conf` and it fixed it.

**Comment by MrYuto** [comment]
I had this weird rendering issue with firefox developer edition after updating hyprland to 0.37.1 and firefox to 125.0b1 without using scaling or `widget.wayland.fractional-scale.enabled` . The funny thing is that enabing the fractional scale at least fixed the blured new window with the traces of the mouse icon that remain until you move focus on the window again or when I was entring my password in bitwarden extension popup inside the browser, that specific screen area was not being updated untill you move the mouse.

**Comment by Kommynct** [comment]
fixed with 0.42.0
